docs/concepts.rst

```diff
@@ -0,0 +1,462 @@
+
+Use of FilterQuery
+''''''''''''''''''
+
+Since the traditional operations did not incorporate the query filters into
+their input parameters, if a query filter is included in the request and the
+request is passed to a traditional operation, the request will be refused
+and an exception generated. This is because the specification for the
+`FilterQuery` states that the server must return filtered responses and
+there is no way to do that with the traditional operations.
+
+Paths in returned instances
+'''''''''''''''''''''''''''
+
+The requirements on paths in returned instances differ between pull and
+traditional operations. The iter operations have been defined to be in line
+with the requirements on paths for pull operations, and the implementation
+of the iter operations acts to bring the path in returned instances in line
+with the requirements of the pull operations, if it uses the traditional
+operation. Thus, the iter operation always returns a complete path in any
+returned instances.
+
+Use of MaxObjectCount argument
+''''''''''''''''''''''''''''''
+
+The `MaxObjectCount` argument is somewhat more limited than if the pull
+operations are used directly in that:
+
+1. It is the same value for open and pull requests.
+2. The mechanism to delay responses (setting `MaxObjectCount=0` and
+   executing a Pull...() method) cannot be used so the interoperation timeout
+   must be sufficient for the client to complete its processing.
+
+Receiving returned objects before an exception
+''''''''''''''''''''''''''''''''''''''''''''''
+
+In general the pull operations receive either objects or error for each
+request (open or pull). Since these operations may be called to get objects
+from the server the iterator may receive objects before an exception is
+executed. In general, unless the `ContinueOnError` flag is set, the
+enumeration sequence will terminate after the first error and that error is
+an indication that not all objects were received from the server. Note that
+if the traditional enumerate function is called by the Iter...() method,
+either objects or an error are received, never both.
+
+Closing a pull operation before it is complete
+''''''''''''''''''''''''''''''''''''''''''''''
+
+An iter operation may be closed before the processing from the server is
+complete by executing the `close()` function on the iterator:
+
+::
+
+    inst_iterator = conn.IterEnumerateInstances(classname,
+                                                MaxObjectCount=max_obj_cnt)
+    for inst in inst_iterator:
+        if <instance fails some test>
+            inst_iterator.close()
+        else:
+            <process the instance>
+
+Note that if the operation executed was the traditional operation rather
+than the pull operation, the `close()` will do nothing since the response
+instances are received as a single block. If the enumeration sequence is
+already complete, this call will also be ignored.
+
+.. _`WBEM Indications`:
```

Comment:
Make link anchor the same as the heading text, except where not possible because of ambiguity.